### PR TITLE
fix yocto shell env

### PR DESCRIPTION
**Pull request by bignaux**

### CODE REVIEW

**Comment by Mic92** [comment]
Looks good to me. Can you add a comment, which version you tested yocto on for future reference?

File: envs/yocto/shell.nix [hunk 1]
```diff
@@ -37,23 +38,23 @@ let
         # TODO limit export to native pkgs?
         nixconf = pkgs.writeText "nixvars.conf" ''
           # This exports the variables to actual build environments
-          # From BB_ENV_EXTRAWHITE
+          # From BB_ENV_PASSTHROUGH_ADDITIONS
```
Comment (otavio) [inline envs/yocto/shell.nix:41]: Could you keep the previous one, as it allows use with older releases?

Comment (bignaux) [reply envs/yocto/shell.nix:41]: ok

**Review by otavio — APPROVED** [review]

**Comment by bignaux** [comment]
The issue now is that : 
```
ERROR: Variable BB_ENV_EXTRAWHITE has been renamed to BB_ENV_PASSTHROUGH_ADDITIONS
ERROR: Variable BB_ENV_EXTRAWHITE from the shell environment has been renamed to BB_ENV_PASSTHROUGH_ADDITIONS
ERROR: Exiting to allow enviroment variables to be corrected
```
i've to unset BB_ENV_EXTRAWHITE . Need to find a workaround.

**Comment by bachp** [comment]
Does anybody need compatibility with older yocto versions? If not I suggest we just move forward with the new settings instead of adding complexity.

**Comment by Mic92** [comment]
> Does anybody need compatibility with older yocto versions? If not I suggest we just move forward with the new settings instead of adding complexity.

sounds good to me.

**Comment by jhol** [comment]
I just used this to built the [OE4T (OpenEmbedded for Tegra) Demo Distro](https://github.com/OE4T/tegra-demo-distro). It all worked in the end.

It works better with `export BB_ENV_EXTRAWHITE="$BB_ENV_PASSTHROUGH_ADDITIONS"` is removed. For this project, I need Kirkstone, and bitbake gives errors if is `BB_ENV_EXTRAWHITE` defined.

There is an issue with the *shadow* recipe trying to pick up *libcrypt* from glibc. However, *libcrypt* has been removed from glibc in recent releases, and a replacement library is available: *libxcrypt*. For some reason, although *libxcrypt* is supported in the project, autotools is still trying to pick up the glibc version. As a hack I replaced `DEPENDS = "virtual/crypt"` with `DEPENDS = "libxcrypt"` in `repos/poky/meta/recipes-extended/shadow/shadow.inc`. I don't know why this is necessary, because to my understanding `virtual/crypt` should be set as an alias of `libxcrypt`.

More investigation is required. I don't know exactly what's going wrong here. I guess the system version of libcrypt leaks on Ubuntu etc., but on Nix it can't be accessed? Anyway, none of this is a nix-environments problems, so I think this branch is generally in good shape apart from the `BB_ENV_EXTRAWHITE` issue.

**Comment by Mic92** [comment]
Ok. Looking forward to a new PR that is fixing any of this. Closing this one because it stalled.

**Pull request closed without merging**